host/greatfet/commands/greatfet_logic.py

```diff
@@ -39,16 +42,52 @@ def main():
         sys.exit(errno.ENODEV)
 
     device.apis.logic_analyzer.start()
-
     time.sleep(1)
 
-    f = open(args.filename, 'wb')
+    fd, path = tempfile.mkstemp(None, None, os.getcwd())
+    if args.output:
+        bin_file_name = args.output
+        arc_name = args.output + '.sr'
+    else:
+        bin_file_name = path[-11:]  # temporary filename is 11 characters
+        arc_name = 'pulseview_session.sr'
+
+    print("Press Ctrl+C to stop reading data from device")
     try:
-        while True:
-            d = device.comms.device.read(0x81, 16384, 1000)
-            f.write(d)
-    except KeyboardInterrupt:
-        pass
+        with open(bin_file_name, "wb") as bin_file, os.fdopen(fd, "wb") as tmp:
+            try:
+                while True:
+                    d = device.comms.device.read(0x81, 16384, 1000)
+                    bin_file.write(d)
+                    tmp.write(d)
+            except KeyboardInterrupt:
+                print()
+
+            if args.output:
+                print("Binary data written to file '%s'" % args.output)
+            if args.pulseview:
+                metadata_str = ("[device 1]\n"
+                                "capturefile=%s\n"
+                                "total probes=8\n"
+                                "samplerate=17 MHz\n"
+                                "total analog=0\n"
+                                "probe1=SGPIO0\n"
+                                "probe2=SGPIO1\n"
+                                "probe3=SGPIO2\n"
+                                "probe4=SGPIO3\n"
+                                "probe5=SGPIO4\n"
+                                "probe6=SGPIO5\n"
+                                "probe7=SGPIO6\n"
+                                "probe8=SGPIO7\n"
+                                "unitsize=1\n") % bin_file_name
+                # pulseview compatible .sr archive
+                with ZipFile(arc_name, "w") as zip:
+                    zip.write(bin_file_name)
+                    zip.writestr("metadata", metadata_str)
+                    zip.writestr("version", "2")
+                print("Pulseview compatible session file created: '%s'" % arc_name)
+    finally:
+        os.remove(path)
```

Comment:
This logic will probably have to change to match the change above-- I'd suggest only running `mkstemp` at all if the pulseview option is set; and then guarding the removal with the same condition.